dvc/repo/fetch.py

```diff
@@ -22,7 +37,36 @@ def fetch(
             remote=remote,
             jobs=jobs,
             recursive=recursive,
-        )["local"]
-        return self.cloud.pull(
-            used, jobs, remote=remote, show_checksums=show_checksums
         )
+
+        downloaded = 0
+        failed = 0
+
+        try:
+            downloaded += self.cloud.pull(
+                used["local"],
+                jobs,
+                remote=remote,
+                show_checksums=show_checksums,
+            )
+        except NoRemoteError:
+            if not used["repo"] and used["local"]:
+                raise
+
+        except DownloadError as exc:
+            failed += exc.amount
+
+        for dep in used["repo"]:
+            try:
+                out = dep.fetch()
+                downloaded += out.get_files_number()
+            except (DownloadError, CloneError, OutputNotFoundError) as exc:
+                if hasattr(exc, "amount"):
```

Comment:
This is ugly :) How about
```
except DownloadError as exc:
    failed += exc.amount
except (CloneError, OutputNotFoundError) as exc:
    failed += 1
```

Reply:
Also, we didn't log `CloneError, OutputNotFoundError` in any way, so they are skipped silently here, which is not right. We should at least log them as we do in `_download` in `Remote`s.

Reply:
Got it :+1: